app/pages/patientdata/patientdata.js

```diff
@@ -526,6 +512,61 @@ export let PatientData = React.createClass({
       datetimeLocation: datetime
     });
   },
+
+  componentWillMount: function() {
+    this.doFetching(this.props);
+    this.doProcessing(this.props);
```

Comment:
Still want to take the `doProcessing` out of here and leave it only in `componentWillReceiveProps` right? It is getting called 8-9 times now for each patient view.